folding-schemes/src/folding/protogalaxy/mod.rs

```diff
@@ -62,51 +67,68 @@ impl<C: CurveGroup> CycleFoldConfig for ProtoGalaxyCycleFoldConfig<C> {
 /// in ProtoGalaxy instances.
 pub type ProtoGalaxyCycleFoldCircuit<C, GC> = CycleFoldCircuit<ProtoGalaxyCycleFoldConfig<C>, GC>;
 
+/// The committed instance of ProtoGalaxy.
+///
+/// We use `RUNNING` to distinguish between incoming and running instances, as
+/// they have slightly different structures (e.g., length of `betas`) and
+/// behaviors (e.g., in satisfiability checks).
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct CommittedInstance<C: CurveGroup> {
+pub struct CommittedInstance<C: CurveGroup, const RUNNING: bool> {
```

Comment:
I don't really like adding the `const_generic` here. This could easily be an enum. In fact, it looks like we use the `RUNNING` flag to not have an `enum`.

Not sure why it was done like this or what pros does it bring. I'm interested TBH. As you have essentially different types. Whereas an enum does the same with more cohesion,

Reply:
I think using const generic here would explicitly distinguish between running and incoming instance at the type level. For instance, in `Folding::prove`, we can be confident that `instance` is indeed a running instance, and `vec_instances` is indeed a vector of incoming instances, thereby avoiding misuses without introducing additional checks.
https://github.com/privacy-scaling-explorations/sonobe/blob/05d9495dd30b3cd7513a57df447f1b1e3f77fc12/folding-schemes/src/folding/protogalaxy/folding.rs#L35-L44

Having said that, enums will definitely make the code more flexible, at the cost of manual shape checks. So both come with tradeoffs, and I don't mind to use an enum if you prefer the latter design :)

Reply:
Hmm I see. 

Maybe we can create two constants like:
`RUNNING` and `INCOMING` and use them instead of a pure `true` or `false`. 
Although the same behind the scenes, it definitely makes it easier to follow the code.


Also, it's true that this makes a type-level diff. And it's positive too. Plus we can always unify behaviour via Traits.
So sure, let's leave as is. But I think makes sense to instanciate with `RUNNING` and `INCOMMING` instead of `true, false`.